lib/get-selenium-status-url.js

```diff
@@ -1,36 +1,71 @@
-module.exports = getSeleniumStatusUrl;
-
 var URI = require('urijs');
+var PROCESS_TYPES = exports.PROCESS_TYPES = {
+  STANDALONE: 0,
+  GRID_HUB: 1,
+  GRID_NODE: 2
+};
 
-function getSeleniumStatusUrl(seleniumArgs) {
-  var port = 4444;
-  var hostname = 'localhost';
-  var portArg = seleniumArgs.indexOf('-port');
-  var hubArg = seleniumArgs.indexOf('-hub');
+exports.getRunningProcessType = function(seleniumArgs) {
   var roleArg = seleniumArgs.indexOf('-role');
+  var role = (roleArg !== -1) ? seleniumArgs[roleArg + 1] : undefined;
 
-  if (hubArg !== -1) {
-    hubURI = new URI(seleniumArgs[hubArg + 1]);
-    hostname = hubURI.hostname() || hostname;
-    port = hubURI.port() || port;
-  }
+  if (roleArg === -1) return PROCESS_TYPES.STANDALONE;
+  else if (role === 'hub') return PROCESS_TYPES.GRID_HUB;
+  else if (role === 'node') return PROCESS_TYPES.GRID_NODE;
+  else return undefined;
+}
+
+exports.getSeleniumStatusUrl = function(seleniumArgs) {
+  var processType = this.getRunningProcessType(seleniumArgs);
+  var portArg = seleniumArgs.indexOf('-port');
+  var port = (portArg !== -1) ? seleniumArgs[portArg + 1] : undefined;
 
-  if (roleArg !== -1) {
-    switch(seleniumArgs[roleArg + 1]){
-      case 'node':
-        if(hubArg === -1){
-          port = portArg !== -1 ? seleniumArgs[portArg + 1] : 5555;
-        }
-        return 'http://' + hostname + ':' + port + '/wd/hub/status';
-      case 'hub':
-        port = portArg !== -1 ? seleniumArgs[portArg + 1] : 4444;
-        return 'http://localhost:' + port + '/api/grid/hub/';
-    }
+  var statusURI = new URI('http://localhost');
```

Comment:
The only way that I can see the hub binding on another IP address or hostname is by using the `-host` parameter. The old code looked at the `-hub` parameter which had no bearing on the binding.

```shell
$ java -jar 2.47.1-server.jar -role hub -hub http://alias:4567/grid/register
...
2015-11-20 11:22:58.075:INFO:osjs.AbstractConnector:Started SocketConnector@0.0.0.0:4444
11:22:58.076 INFO - Nodes should register to http://153.32.222.55:4444/grid/register/
11:22:58.076 INFO - Selenium Grid hub is up and running
```

```shell
$ java -jar 2.47.1-server.jar -role hub -host alias -port 4567
...
2015-11-20 11:14:22.961:INFO:osjs.AbstractConnector:Started SocketConnector@alias:4567
11:14:22.964 INFO - Nodes should register to http://alias:4567/grid/register/
11:14:22.964 INFO - Selenium Grid hub is up and running
```

I didn't know about the `-host` parameter before, so I'll add support for that instead of hardcoding localhost.

---
Found the list of supported command line arguments for the `node` and `hub` roles in https://github.com/SeleniumHQ/selenium/blob/selenium-2.48.2/java/server/src/org/openqa/grid/common/defaults/NodeOptions.json and https://github.com/SeleniumHQ/selenium/blob/selenium-2.48.2/java/server/src/org/openqa/grid/common/defaults/HubOptions.json